### PR TITLE
Enhanced Stop Viewer Bubbles

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
This PR brings enhanced stop viewer bubbles, "completing" the stop viewer enhancements.

Screen Shot 2021-10-13 at 3 57 32 PM

There are some strange things going on with the placement of the caret at the bottom of the bubble. If anyone has an idea of how to improve this, I would greatly appreciate any help there!

### CODE REVIEW

**Comment by daniel-heppner-ibigroup** [comment]
I've noticed that if I click somewhere random on the map and open an address popover, then click on the enhanced stop bubble, it opens another address popover for the address behind the bubble rather than the popover for that stop. But if the last thing I clicked was another stop, then I can click anywhere on the enhanced stop bubble and get information for that stop rather than the address behind the bubble.

File: lib/components/map/enhanced-stop-marker.js [hunk 1]
```diff
@@ -52,29 +76,41 @@ class EnhancedStopMarker extends Component {
...
-    if (!stopLabel) return null
-    const width = getTextWidth(stopLabel) * 0.83 + 80 / getTextWidth(stopLabel)
+    if (!stopCode) return null
+
+    const mode = getModeFromStop(stop)
+    let color = modeColors && modeColors[mode] ? modeColors[mode] : 'pink'
+    if (hovered) {
+      // Generates a pretty variant of the color
+      color = tinycolor(color).monochromatic()[3].toHexString()
+    }
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply lib/components/map/enhanced-stop-marker.js]: Have renamed it to `highlight` in babb22ee345fd3c6771e7ef66c3130099cd0aab6 to make it more agnostic to where it is highlighted from.

**Comment by miles-grant-ibigroup** [comment]
Thank you @binh-dam-ibigroup for the PR fix! Merged into this branch.

@daniel-heppner-ibigroup that is a very good find thank you! I think the auto-zoom functionality is definitely important, but out of scope for this PR. Have added it to #483 which contains all the things we should fix following the enhanced stop viewer PRs.

The issue with reverse geocoding breaking the click behavior is a good catch. It looks like it's also happening with the bike share overlay which to me suggests it's a larger bug not related to this PR. I've created issue #485 which I would close out separately since it looks like this issue wasn't created by this PR.

**Comment by daniel-heppner-ibigroup** [comment]
So, @miles-grant-ibigroup, are we not going to try to fix the click behavior issue right now?

**Review by daniel-heppner-ibigroup — APPROVED** [review]
Looks good, aside from bugs I pointed out which it sounds like we will not be fixing in this PR.

File: lib/util/viewer.js [hunk 2]
```diff
@@ -159,6 +159,40 @@ export function getModeFromRoute (route) {
...
+        return modes
+      }, {})
+
+    // Get stop mode by getting most common stop mode
+    return stopModeCounts && Object.keys(stopModeCounts)
+      // Sort by mode occurrence
+      .sort((a, b) => {
+        return stopModeCounts[a] - stopModeCounts[b]
+      })[0]
+  }
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply lib/util/viewer.js:193]: Thank you for checking! I think as long as it doesn't crash we're good for now.

**Comment by miles-grant-ibigroup** [comment]
@daniel-heppner-ibigroup Yes I think the click behavior changes belong in a separate PR.

**Pull request merged**